rules/manager.go

```diff
@@ -850,6 +856,10 @@ func (g *Group) Equals(ng *Group) bool {
 		return false
 	}
 
+	if g.limit != ng.limit {
+		return false
```

Comment:
do we test this?

Reply:
Yes https://github.com/prometheus/prometheus/blob/e5c8575188f00dc7f6a53b884828559efebb5011/rules/manager_test.go#L776-L780